### PR TITLE
 add option to select backends TF/PT 

### PR DESCRIPTION
**Pull request by thangckt**
reopen PR #1541 due to branch is deleted

add a new key in `param.json` file

```
"train_backend": "pytorch"/"tensorflow",
```
relate to this issue https://github.com/deepmodeling/dpgen/issues/1462

## Summary by CodeRabbit

- **New Features**
	- Improved model management by dynamically generating model suffixes based on the selected backend, enhancing compatibility.
  
- **Enhancements**
	- Updated model-related functions to incorporate backend-specific model suffixes for accurate file handling during training processes.

### CODE REVIEW

File: dpgen/generator/run.py [hunk 1]
```diff
@@ -730,7 +748,13 @@ def run_train(iter_index, jdata, mdata):
             "training_init_model, training_init_frozen_model, and training_finetune_model are mutually exclusive."
         )
 
-    train_command = mdata.get("train_command", "dp")
+    train_command = mdata.get("train_command", "dp").strip()
+    # assert train_command == "dp", "The 'train_command' should be 'dp'"     # the tests should be updated to run this command
+    if suffix == ".pb":
+        train_command += " --tf"
```
Comment (njzjz) [inline dpgen/generator/run.py:754]: This breaks support for old versions. This line can be removed, as `--tf` is the default behavior.

File: dpgen/generator/run.py [hunk 2]
```diff
@@ -2177,7 +2220,7 @@ def _read_model_devi_file(
         assert all(
             model_devi_content.shape[0] == model_devi_contents[0].shape[0]
             for model_devi_content in model_devi_contents
-        ), r"Not all beads generated the same number of lines in the model_devi${ibead}.out file. Check your pimd task carefully."
+        ), "Not all beads generated the same number of lines in the model_devi$\{ibead\}.out file. Check your pimd task carefully."
```
Comment (njzjz) [inline dpgen/generator/run.py:2223]: This change should be reverted.

**Review by njzjz — CHANGES_REQUESTED** [review]
It looks good to me. Two minor changes should be made.

File: dpgen/generator/run.py [hunk 3]
```diff
@@ -798,17 +822,26 @@ def run_train(iter_index, jdata, mdata):
             os.path.join("old", "model.ckpt.data-00000-of-00001"),
         ]
```
Comment (thangckt) [inline dpgen/generator/run.py:823]: The file `model.ckpt.pt` is collected, but not being used. I guess it should be used around here. However, I am not familiar with this process then I can not  make a change. Hope you check this.

File: dpgen/generator/run.py (hunk 2 above)
Comment (thangckt) [reply dpgen/generator/run.py:2223]: This change is made by your bot, not me.

Comment (njzjz) [reply dpgen/generator/run.py]: It's made by https://github.com/deepmodeling/dpgen/commit/8f7f491bfa6d850d344267e0f3928aca6ae90b46

**Review by njzjz — APPROVED** [review]
It looks good to me. I expect someone to test it (I don't have time to do so)

**Comment by Yi-FanLi** [comment]
I have just tested it on my system. This PR works fine itself, but the PyTorch model cannot be correctly trained due to the inconsistency with the TensorFlow backend. See https://github.com/deepmodeling/deepmd-kit/issues/3770 and https://github.com/deepmodeling/deepmd-kit/issues/3771.

**Pull request merged**